### PR TITLE
Add P3 flight segments

### PR DESCRIPTION
**Pull request by RobertPincus**
Early and incomplete but dropsonde circle times are present and Geet George should be able to move forward with JOANNE.

### CODE REVIEW

**Comment by RobertPincus** [comment]
@d70-t I'm not sure why the CI isn't running on this PR. 

@Geet-George These files will get revised but, at long last, here are accurate timings for the dropsonde circles from P3

**Comment by RobertPincus** [comment]
@Geet-George @d70-t I updated the `sondes.yaml` file with what Geet provided in the Mattermost channel. Most of the P3 segment files are passing (I have to fine-tune some times), but the sondes in the fifth HALO flight segmentation  file (0128) are inconsistent with the new `sondes.yaml`

**Comment by RobertPincus** [comment]
@Geet-George @d70-t Sigh. Now my P3 files are working but the HALO file validation is a mess. Has something changed at AERIS, in addition to whatever has happened to the sondes?

File: scripts/verify.py [hunk 1]
```diff
@@ -36,10 +36,13 @@ def validate(segment_file, sonde_info):
...
         warnings = list(checker.check_segment(seg, seg_navdata, sondes_by_flag))
         for warning in warnings:
-            segmentlogger.warn(warning)
+            if "segment_id" in seg:
+                segmentlogger.warning(seg["segment_id"])
+            segmentlogger.warning(warning)
 
         segment_warning_count += len(warnings)
 
+    navdata.close()
```
Comment (d70-t) [inline scripts/verify.py]: This is a very good addition. However, I'd rather change that to be using the [`closing` context manager](https://docs.python.org/3/library/contextlib.html#contextlib.closing) e.g.:
```python
from contextlib import closing
...
with closing(get_navdata(...).load()) as navdata:
    ....
```

File: scripts/sondes.yaml [hunk 2]
```diff
@@ -1254,19 +1254,19 @@
...
   platform: HALO
   sonde_id: HALO-0130_s02
-- flag: GOOD
+- flag: BAD
   launch_time: 2020-01-30 13:08:32
   platform: HALO
   sonde_id: HALO-0130_s03
-- flag: GOOD
+- flag: BAD
   launch_time: 2020-01-30 13:35:57
```
Comment (d70-t) [inline scripts/sondes.yaml]: This changes four HALO dropsondes from `GOOD` to `BAD`. That's maybe not somethig one would expect from the addition of P3 segments. Why is this happening? Should that go into a separate pull request?

File: scripts/requirements.txt [hunk 3]
```diff
@@ -6,7 +6,7 @@ xarray
 requests
 aiohttp
 intake
-git+https://github.com/intake/intake-xarray@master#egg=intake-xarray
+intake-xarray
```
Comment (d70-t) [inline scripts/requirements.txt]:
```suggestion
intake-xarray>=0.3.2
```

**Comment by d70-t** [comment]
> but the HALO file validation is a mess

The opendap Server at Aeris has some problems... This kind of error appears from time to time and currently probably more often than not.

File: scripts/sondes.yaml (hunk 2 above)
Comment (RobertPincus) [reply scripts/sondes.yaml]: @Geet-George This happened as part of a change to your `sondes.yaml` file. Should I change them back as part of my pull request?

File: scripts/verify.py (hunk 1 above)
Comment (RobertPincus) [reply scripts/verify.py]: Excellent idea, I did this.

File: scripts/verify.py [hunk 4]
```diff
@@ -66,7 +66,8 @@ def _main():
     sonde_info = yaml.load(open(args.sonde_info), Loader=yaml.SafeLoader)
 
     total_warnings = 0
-    for filename in tqdm.tqdm(args.infiles):
+    # for filename in tqdm.tqdm(args.infiles):
+    for filename in args.infiles:
```
Comment (d70-t) [inline scripts/verify.py]: Is this a temporary fix which slipped in? Or should tqdm be removed / made optional?

Comment (RobertPincus) [reply scripts/verify.py]: <facepalm> I try to keep my local environments small so made this change for my local testing, sorry. I reverted the change.

Comment (d70-t) [reply scripts/verify.py]: Probably tqdm should be optional then... But that's for a separate PR. Thanks for reverting it.

**Review by d70-t — APPROVED** [review]
Look good, all green now :-)

File: scripts/sondes.yaml (hunk 2 above)
Comment (Geet-George) [reply scripts/sondes.yaml]: Those changes are a mistake on my end, I think. They should remain as `GOOD`, and not change to `BAD`. I can reupload the new `sondes.yaml` file.

**Comment by Geet-George** [comment]
Thanks a lot, @RobertPincus for the segment files of the P3. They will be very helpful for me. Will start on them today itself. :)

File: scripts/sondes.yaml (hunk 2 above)
Comment (d70-t) [reply scripts/sondes.yaml]: If they should stay the same, everything is fine. Robert removed the change from the PR before we merged it.

Comment (Geet-George) [reply scripts/sondes.yaml]: Yeah... Just saw the commit for that. Then, everything is good. I am not changing/uploading anything.
